test/WebJobs.Script.Tests/NodeEndToEndTests.cs

```diff
@@ -734,6 +734,72 @@ public void ExcludedFunction_NotAddedToHost()
             }
         }
 
+        [Fact]
+        public async Task PromiseApi_Resolves()
+        {
+            JObject input = new JObject
+            {
+                { "scenario", "promiseApiResolves" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
+
+            Task result = await Task.WhenAny(t, Task.Delay(5000));
+            Assert.Same(t, result);
+            if (t.IsFaulted)
+            {
+                throw t.Exception;
+            }
+        }
+
+        [Fact]
+        public async Task PromiseApi_Rejects()
+        {
+            JObject input = new JObject
+            {
+                { "scenario", "promiseApiRejects" }
+            };
+
+            Task t = Fixture.Host.CallAsync("Scenarios",
+                new Dictionary<string, object>()
+                {
+                    { "input", input.ToString() }
+                });
+
+            Task result = await Task.WhenAny(t, Task.Delay(5000));
+            Assert.Same(t, result);
+            Assert.Equal(true, t.IsFaulted);
+            Assert.Equal("reject", t.Exception.InnerException.InnerException.Message);
```

Comment:
Is this dealing with nested `AggregateExceptions` here? If so, just use `Flatten` to clean this up a bit.